### PR TITLE
Fix/xy plot npe silent data loss

### PR DESCRIPTION
**Pull request by Pradyuman-aviator**
Describe changes proposed in this pull request:
- Fixed a `NullPointerException` in `convertPatientClinicalDataToSampleClinicalData`: the chained call `patientToSamples.get(patientId).get(studyId)` crashed when a patient had clinical data but no entry in the filtered sample map
- Fixed silent data loss: when `samplesForPatient` was null, the patient's XY plot data point was silently dropped with no log or error  researchers could unknowingly see incomplete XY plots
- Both edge cases now emit a structured `log.warn()` so operators can detect data integrity issues
- Added `GetClinicalDataForXyPlotUseCaseTest` - first test coverage for this class - with 6 cases covering the happy path, multi-sample patients, both bug scenarios, sample-only passthrough, and empty-value filtering

# Checks
- [x] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/)
- [x] Has tests: `GetClinicalDataForXyPlotUseCaseTest` - 6 tests, all passing
- [x] No clinical attribute logic added -this is a data routing fix only
- [x] Label: `bug`

# Notify reviewers
did  blame...it shows this file was introduced by @fuzhaoyuan  in #11425 tagging as reviewer

### CODE REVIEW

**Comment by fuzhaoyuan** [comment]
Thanks! Will take a look

**Comment by alisman** [comment]
@Pradyuman-aviator thanks for this PR.  We're worried this might bloat logs with a condition that is sometimes legitimate.   I think that if this just fails silently it will not affect the integrity of the analysis itself?

File: src/main/java/org/cbioportal/domain/clinical_data/usecase/GetClinicalDataForXyPlotUseCase.java [hunk 1]
```diff
@@ -137,26 +141,43 @@ private List<ClinicalData> convertPatientClinicalDataToSampleClinicalData(
...
-        // return new ResponseEntity<>(null, HttpStatus.INTERNAL_SERVER_ERROR);
+      Map<String, List<Sample>> samplesByStudy = patientToSamples.get(d.patientId());
+      if (samplesByStudy == null) {
+        log.warn(
+            "Patient '{}' has patient-level clinical data (attribute '{}') but no samples "
+                + "in the filtered set — skipping XY plot data point. "
+                + "This may indicate a data integrity issue.",
+            d.patientId(),
+            d.attrId());
+        continue;
```
[the remark this answers is not in the document]
Comment (Pradyuman-aviator) [reply src/main/java/org/cbioportal/domain/clinical_data/usecase/GetClinicalDataForXyPlotUseCase.java:149]: Addressed and taken a step further Initially replaced per-record 
log.warn calls with counters + a single summary warn after the loop 
(as suggested). However, @alisman pointed out in review that this skip 
is a legitimate state when samples are filtered out by the study view 
filter, not a data integrity issue. Removed all logging entirely  
the fix is now purely a null-safety improvement to prevent the NPE.

File: src/test/java/org/cbioportal/domain/clinical_data/usecase/GetClinicalDataForXyPlotUseCaseTest.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  @Mock private GetSampleClinicalDataUseCase getSampleClinicalDataUseCase;
+  @Mock private GetFilteredSamplesUseCase getFilteredSamplesUseCase;
+  @Mock private StudyViewFilterContext filterContext;
+
+  private GetClinicalDataForXyPlotUseCase useCase;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.openMocks(this);
+    useCase =
```
[the remark this answers is not in the document]
Comment (Pradyuman-aviator) [reply src/test/java/org/cbioportal/domain/clinical_data/usecase/GetClinicalDataForXyPlotUseCaseTest.java:37]: FIxed

**Comment by Pradyuman-aviator** [comment]
Thanks for the clarification @alisman! Makes sense  when samples 
are filtered out by the study view filter, the skip is expected behaviour, 
not a data integrity issue. Removed all logging. The fix is now purely 
null-safe lookups to prevent the NPE, preserving the original silent-skip 
behaviour. Updated the branch.
